tests/unit/test_credentials.py

```diff
@@ -1716,6 +1718,17 @@ def test_custom_cache(self):
         cache = resolver.get_provider('assume-role').cache
         self.assertIs(cache, custom_cache)
 
+    @mock.patch('botocore.awsrequest.AWSRequest')
+    def test_imds_use_truncated_user_agent(self, request):
+        session = Session()
```

Comment:
Does it not work to use the pattern the prior tests use where we pull out a provider using 

```python
        resolver = credentials.create_credential_resolver(self.session)
        provider = resolver.get_provider('iam-role')
        provider.load()
```

I would imagine we shouldn't have to mock out the awsrequest and instead be able to inject mocks.